File: 프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

bool check[50005];
int answer;

bool compare(int a, int b) {
    return a > b;
}

int solution(vector<int> people, int limit) {
    fill(check, check+50005, false);
    sort(people.begin(), people.end(), compare);
    
    for(int i=0; i<people.size(); i++) {
        if(!check[i] && people[i] <= limit) {
            check[i] = true;
            int tmplimit = limit - people[i];
            for(int j=i+1; j<people.size(); j++) {
                if(tmplimit - people[j] >= 0) {
                    tmplimit -= people[j];
                    check[j] = true;
                }
            }
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int start = 0;
        int ans = 0;
        for(int i=people.length-1; i>=start; i--) {
            if(people[i] + people[start] <= limit) {
                start++;
                ans++;
            }
            else ans++;
            // 50 50 70 80
        }
        return ans;
    }
}